meshtastic/tests/test_mesh_interface_behavioral_compat.py

```
@@ -0,0 +1,1851 @@
+    # -------------------------------------------------------------------------
+    # 7. sendTelemetry edge cases
+    # -------------------------------------------------------------------------
+
+    def test_sendTelemetry_with_invalid_type_logs_warning(self, mock_interface, caplog):
+        """Test that sendTelemetry with invalid type logs a warning and falls back."""
+        iface = mock_interface
+
+        with caplog.at_level(logging.WARNING):
+            with patch.object(
+                iface._send_pipeline, "_send_data_with_wait"
+            ) as mock_send:
```

Ruff (0.15.7)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

Combine `with` statements

(SIM117)